app/controllers/admin/statistics_announcements_controller.rb

```diff
@@ -71,6 +73,19 @@ def build_statistics_announcement(attributes = {})
     current_user.statistics_announcements.new(attributes)
   end
 
+  def modified_statistics_announcement_params
+    if params&.dig(:statistics_announcement, :current_release_date_attributes, :precision) == "exact_confirmed"
+      confirm_exact_statistics_announcement
+    end
+
+    statistics_announcement_params
+  end
+
+  def confirm_exact_statistics_announcement
+    params[:statistics_announcement][:current_release_date_attributes][:precision] = StatisticsAnnouncementDate::PRECISION[:exact]
+    params[:statistics_announcement][:current_release_date_attributes][:confirmed] = true
+  end
```

Comment:
optional: We've got a pattern in other controllers of munging params via before action. If we do that this would look something like:

```
class Admin::StatisticsAnnouncementsController < Admin::BaseController
before_action :set_release_date_params
```

```
def statistics_announcement_params
    @statistics_announcement_params ||= params
      .require(:statistics_announcement)
      .permit(
        :title,
        :summary,
        :publication_type_id,
        :publication_id,
        :cancellation_reason,
        organisation_ids: [],
        topic_ids: [],
        current_release_date_attributes: %i[id release_date precision confirmed],
      )
  end

def set_release_date_params
  if params&.dig(:statistics_announcement, :current_release_date_attributes, :precision) == "exact_confirmed"
      params[:statistics_announcement][:current_release_date_attributes][:precision] = 
      StatisticsAnnouncementDate::PRECISION[:exact]
     params[:statistics_announcement][:current_release_date_attributes][:confirmed] = true
  else
      params[:statistics_announcement][:current_release_date_attributes][:confirmed] = true
  end
end

```